xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -105,24 +105,38 @@ bool CPVRTimers::IsRecording(void) const
 
 bool CPVRTimers::SetEpgTagTimer(const CPVRTimerInfoTagPtr &timer)
 {
-  const CEpgInfoTagPtr epgTag(timer->GetEpgInfoTag());
-  if (epgTag)
-  {
-    epgTag->SetTimer(timer);
-    return true;
-  }
-  return false;
+  if (timer->m_bStartAnyTime || timer->m_bEndAnyTime)
+    return false;
+
+  std::vector<CEpgInfoTagPtr> tags(g_EpgContainer.GetEpgTagsForTimer(timer));
+
+  if (tags.empty())
+    return false;
+
+  // assign first matching epg tag to the timer.
+  timer->SetEpgTag(tags.front());
```

Comment:
Does the timer have to have an EPG tag associated? If we could skip this we could avoid the potential locking issues introduced by `SetEpgTag()`.